server/src/utils/protocol/attributes/kvstore/kvstore.put.ts

```diff
@@ -0,0 +1,42 @@
+import { Address } from "@liftedinit/many-js";
+import { parseAddress } from "../../../cbor-parsers";
+import { Analyzer } from "../../analyzer";
+
+interface ArgumentT {
+  key: string;
+  value: string;
+  owner: string;
+}
+
+type ResultT = null;
+
+interface EventT {
+  key: string;
+  value: string;
+  owner: string;
+}
+
+export class KvstorePutAnalyzer extends Analyzer<ArgumentT, ResultT, EventT> {
+  static method = "kvstore.put";
+  static eventType = [7, 0];
+
+  parseArgs(sender: Address, payload: Map<any, any>): ArgumentT {
+    return {
+      key: payload.get(0),
+      value: payload.get(1),
+      owner: (parseAddress(payload.get(2)) || sender).toString(),
```

Comment:
I don't think that's right, see https://github.com/many-protocol/specification/blob/main/attributes/network/cddl/7_kvstore_commands.cddl